### ISSUE TITLE
Normalize Quaternions

### ISSUE THREAD

**Issue by v4hn** [ros-planning/moveit#1119]
The PlanningScene class does not always correctly normalize quaternions in new collision objects.

Here, the unnormalized w-component is normalized correctly:
```
$ rostopic pub -1 /collision_object moveit_msgs/CollisionObject "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: 'world'
id: 'box'a
primitives:
- type: 1
  dimensions: [1.8,1.5,1.0]
primitive_poses:
- position: {x: 0.0, y: 0.0, z: 0.15}
  orientation: {x: 0.0, y: 0.0, z: 0.0, w: 2.0}"

$ rosservice call /get_planning_scene "components:
  components: 16" | grep box -A20
        id: "box"
        type:
          key: ''
          db: ''
        primitives:
          -
            type: 1
            dimensions: [1.8, 1.5, 1.0]
        primitive_poses:
          -
            position:
              x: 0.89
              y: 0.74
              z: 0.93
            orientation:
              x: 0.0
              y: 0.0
              z: 0.0
              w: 1.0
        meshes: []
        mesh_poses: []
```

But when another component is set, this suddenly fails:
```
$ rostopic pub -1 /collision_object moveit_msgs/CollisionObject "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: 'world'
id: 'box'
primitives:
- type: 1
  dimensions: [1.8,1.5,1.0]
primitive_poses:
- position: {x: 0.0, y: 0.0, z: 0.15}
  orientation: {x: 0.0, y: 15.0, z: 0.0, w: 2.0}"

$ rosservice call /get_planning_scene "components:
  components: 16" | grep box -A20
        id: "box"
        type:
          key: ''
          db: ''
        primitives:
          -
            type: 1
            dimensions: [1.8, 1.5, 1.0]
        primitive_poses:
          -
            position:
              x: 0.89
              y: 0.74
              z: 0.93
            orientation:
              x: 0.0
              y: 15.0
              z: 0.0
              w: 2.0
        meshes: []
        mesh_poses: []
```

**Comment by ggupta9777** [comment]
I'll take this up.

**Comment by bdenoun** [comment]
I was  already working on it, sorry I forgot to assign it to myself :/

**Comment by davetcoleman** [comment]
I'm removing the assigned tag. Did any progress ever get made on this?

**Comment by j-petit** [comment]
I am taking this up. I have already found the cause in the src-files. However, I am unsure about the contribution process, as I am new to moveit!.

My environment is melodic on Ubuntu 18.04. Should I therefore make the PR to be merged into the ``melodic-devel`` branch? And should the new branch where I commit my changes be then also a copy of my forked ``melodic-devel`` branch?

**Comment by rhaschke** [comment]
@j-petit, thanks for handling this issue. 
As this fix will (hopefully) not change API or ABI, it should go into melodic-devel and then be back-ported to Kinetic and forward-ported into master.
So, please branch of from the melodic-devel branch of your clone.

**Comment by rhaschke** [comment]
Fixed via #1420, #1433
